File: services/constant/src/main/java/com/hashjosh/constant/document/dto/DocumentResponse.java
package com.hashjosh.constant.document.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.databind.JsonNode;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class DocumentResponse{
    UUID documentId;
    UUID uploadedBy;
    String fileName;
    String fileType;
    Long fileSize;
    String objectKey;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSSSS")
    LocalDateTime uploadedAt;
    String preview;
}

File: frontend/mobile/lib/data/models/document_response.dart
class DocumentResponse {
  final String documentId;
  final String referenceId;
  final String uploadedBy;
  final String fileName;
  final String fileType;
  final int? fileSize;
  final String objectKey;
  final Map<String, dynamic>? metaData;
  final String uploadedAt;
  final String preview;

  DocumentResponse({
    required this.documentId,
    required this.referenceId,
    required this.uploadedBy,
    required this.fileName,
    required this.fileType,
    this.fileSize,
    required this.objectKey,
    this.metaData,
    required this.uploadedAt,
    required this.preview,
  });

  factory DocumentResponse.fromJson(Map<String, dynamic> json) {
    return DocumentResponse(
      documentId: json['documentId']?.toString() ?? '',
      referenceId: json['referenceId']?.toString() ?? '',
      uploadedBy: json['uploadedBy']?.toString() ?? '',
      fileName: json['fileName'] ?? '',
      fileType: json['fileType'] ?? '',
      fileSize: json['fileSize'],
      objectKey: json['objectKey'] ?? '',
      metaData: json['metaData'] as Map<String, dynamic>?,
      uploadedAt: json['uploadedAt'] ?? '',
      preview: json['preview'] ?? '',
    );
  }
}
